mmpose/evaluation/metrics/keypoint_partition_metric.py

```diff
@@ -0,0 +1,132 @@
+            arguments required by the metric of your choice.
+        partitions (dict): definition of body partitions. For example, if we
+            have 10 keypoints in total, the first 7 keypoints belong to body
+            and the last 3 keypoints belong to foot, this field can be like
+            this:
+                dict(
+                    body=[0, 1, 2, 3, 4, 5, 6],
+                    foot=[7, 8, 9],
+                    all=[0, 1, 2, 3, 4, 5, 6, 7, 8, 9]
+                )
+            where the numbers are the indices of keypoints and they can be
+            discontinuous.
+    """
+
+    def __init__(
+        self,
+        metric: dict,
+        partitions: dict,
+    ) -> None:
+        super().__init__()
+        self.partitions = partitions
+        # instantiate metrics for each partition
+        self.metrics = {
+            partition_name: METRICS.build(metric)
+            for partition_name in partitions.keys()
+        }
+
+    @property
+    def dataset_meta(self) -> Optional[dict]:
+        """Optional[dict]: Meta info of the dataset."""
+        return self._dataset_meta
+
+    @dataset_meta.setter
+    def dataset_meta(self, dataset_meta: dict) -> None:
+        """Set the dataset meta info to the metric."""
+        self._dataset_meta = dataset_meta
+        # sigmas required by coco metric have to be split as well
+        for partition_name, keypoint_ids in self.partitions.items():
+            _dataset_meta = deepcopy(dataset_meta)
+            _dataset_meta['num_keypoints'] = len(keypoint_ids)
+            _dataset_meta['sigmas'] = _dataset_meta['sigmas'][keypoint_ids]
+            self.metrics[partition_name].dataset_meta = _dataset_meta
+
+    def process(self, data_batch: Sequence[dict],
+                data_samples: Sequence[dict]) -> None:
+        """Split data samples by partitions, then call metric.process part by
+        part."""
+        parted_data_samples = {
+            partition_name: []
+            for partition_name in self.partitions.keys()
+        }
+        for data_sample in data_samples:
+            for partition_name, keypoint_ids in self.partitions.items():
+                _data_sample = deepcopy(data_sample)
+                _data_sample['pred_instances'][
+                    'keypoint_scores'] = _data_sample['pred_instances'][
+                        'keypoint_scores'][:, keypoint_ids]
+                _data_sample['pred_instances']['keypoints'] = _data_sample[
+                    'pred_instances']['keypoints'][:, keypoint_ids]
+                _data_sample['gt_instances']['keypoints'] = _data_sample[
+                    'gt_instances']['keypoints'][:, keypoint_ids]
+                _data_sample['gt_instances'][
+                    'keypoints_visible'] = _data_sample['gt_instances'][
+                        'keypoints_visible'][:, keypoint_ids]
+
+                # for coco metric
+                if 'raw_ann_info' in _data_sample:
```

Comment:
Is this wrapper only compatible with CocoMetric instantiated without `ann_file`?

Reply:
Yes, I forgot to mention this.

Reply:
@Ben-Louis I think this partition metric is mainly designed for custom/industrial applications, so it could be acceptable. @ly015 What do you think?

Reply:
I agree. I'm considering the situation where users provide an `ann_file`. Would it be better to raise an AssertionError or simply neglect it with a warning? @Tau-J